docs/other-topics/transactions.md

```diff
@@ -222,6 +222,84 @@ const sequelize = new Sequelize('sqlite::memory:', {
 
 **Note for MSSQL:** _The `SET ISOLATION LEVEL` queries are not logged since the specified `isolationLevel` is passed directly to `tedious`._
 
+## Manually retrying failed transactions
+
+Sequelize does not automically retries on potential write conflicts or deadlocks, when using a certain level of isolations, it is expected that transactions may fail due to potential write conflicts in concurrent transactions or deadlocks.
+
+You can programmatically fix this by relying on options from the [retry-as-promised](https://github.com/mickhansen/retry-as-promised/blob/master/README.md) library:
+
+```typescript
+const { Sequelize } = require('@sequelize/core');
+
+const sequelize = new Sequelize(process.env.DB_NAME, 
+    process.env.DB_NAME, 
+    process.env.DB_PASS, 
+    {
+    host: process.env.DB_HOST,
+    dialect: 'mysql',
+    // Here:
+    retry: {
+        max: 3, // this may cause memory leaks if you set the retries limit too much.
+        match: [
+          Sequelize.ConnectionError,
+          Sequelize.ConnectionRefusedError,
+          Sequelize.ConnectionTimedOutError,
+          Sequelize.OptimisticLockError,
```

Comment:
Regarding retrying on exceptions of type `Sequelize.OptimisticLockError`, does sequelize automagically increment the version number on a model instance for the next retry? Otherwise wouldn't we expect these retries to fail for the same reason, a mismatched version?

Reply:
> Regarding retrying on exceptions of type `Sequelize.OptimisticLockError`, does sequelize automagically increment the version number on a model instance for the next retry? Otherwise wouldn't we expect these retries to fail for the same reason, a mismatched version?

I'm not sure about that, will try looking at it soon

Reply:
> Regarding retrying on exceptions of type `Sequelize.OptimisticLockError`, does sequelize automagically increment the version number on a model instance for the next retry? Otherwise wouldn't we expect these retries to fail for the same reason, a mismatched version?

I'm not sure about that, will try looking on it soon

Reply:
> Regarding retrying on exceptions of type `Sequelize.OptimisticLockError`, does sequelize automagically increment the version number on a model instance for the next retry? Otherwise wouldn't we expect these retries to fail for the same reason, a mismatched version?

 After checking it for a while, unfortunately we don't, but if you are interested to help, that's gonna be nice. I will add the changes about this.

Reply:
> > Regarding retrying on exceptions of type `Sequelize.OptimisticLockError`, does sequelize automagically increment the version number on a model instance for the next retry? Otherwise wouldn't we expect these retries to fail for the same reason, a mismatched version?
> 
> I'm not sure about that, will try looking on it soon

 After checking it for a while, unfortunately we don't, but if you are interested to help adding the mechanism, that's would be nice. I will add the changes about this.